kpops/components/base_components/models/from_section.py

```diff
@@ -16,26 +16,53 @@ class InputTopicTypes(str, Enum):
 
     INPUT = "input"
     EXTRA = "extra"
+    PATTERN = "pattern"
     INPUT_PATTERN = "input-pattern"
     EXTRA_PATTERN = "extra-pattern"
 
 
 class FromTopic(BaseModel):
     """Input topic
 
-    :param type: Topic type
-    :type type: InputTopicTypes
-    :param role: Custom identifier belonging to a topic, provide only if `type` is `extra` or `extra-pattern`
-    :type role: str | None
+    :param type: Topic type, defaults to None
+    :type type: InputTopicTypes | None, optional
+    :param role: Custom identifier belonging to a topic, provide only if `type` is `extra` or `extra-pattern`.
+        When `role` is not `None`, `type: pattern` is equal to `type: extra-pattern`, defaults to None
```

Comment:
perhaps easier to understand?
```suggestion
        When `role` is defined, `type: pattern` is equal to `type: extra-pattern`, defaults to None
```